### ISSUE TITLE
 pilot report warning:"got nil value for route 0.0.0.0" and pod can not work well

### ISSUE THREAD

**Issue by wansuiye** [istio/istio#13695]
**Bug description**
when the pod is injected with proxy, the pilot report warning and pod can not be working well. 

```
# kubectl get pod -n master
NAME                           READY     STATUS    RESTARTS   AGE
user-59dbc5c585-bpzg5      1/2       Running   6          1h
```
the pilot log:
```
2019-04-28T14:14:49.064519Z	info	ads	LDS: PUSH for node:user-c559bf4ff-brcvf.master addr:"10.201.18.250:33930" listeners:156 317519
2019-04-28T14:14:49.064578Z	warn	ads	RDS: got nil value for route 0.0.0.0 for node &amp;{ sidecar [10.201.18.250] user-c559bf4ff-brcvf.master  master.svc.cluster.local master  map[POD_NAME:user-c559bf4ff-brcvf CONFIG_NAMESPACE:master ISTIO_VERSION:1.1.3 pod-template-hash:711569099 ISTIO_PROXY_SHA:istio-proxy:a169a0c0cd86b51538c240e2d037fa8f7f5860ae ISTIO_META_INSTANCE_IPS:10.201.18.250,10.201.18.250 istio:sidecar INTERCEPTION_MODE:REDIRECT ISTIO_PROXY_VERSION:1.1.3 app:user] 0xc421f32190 [] [app-index=1,app=user,m-app=user,pod-template-hash=711569099]}: &lt;nil&gt;
2019-04-28T14:14.49.1629151Z	info	ads	ADS: RDS: PUSH for node: user-c559bf4ff-brcvf.master addr:10.201.18.250:37856 routes:130
2019-04-28T14:14.49.1629201Z   info	ads	Push finished: 1.515309241s {
    "ProxyStatus": {
        "pilot_no_ip": {
            "user-c559bf4ff-brcvf.master": {
                "proxy": "user-c559bf4ff-brcvf.master"
            },
...
```
the deployment
```
{
  "kind": "Deployment",
  "apiVersion": "extensions/v1beta1",
  "metadata": {
    "name": "user",
    "namespace": "master",
    "labels": {
      "app": "user"
    }
  },
  "spec": {
    "replicas": 1,
    "selector": {
      "matchLabels": {
        "app": "user"
      }
    },
    "template": {
      "metadata": {
        "labels": {
          "app": "user"
        }
      },
      "spec": {
        "containers": [
          {
            "name": "user",
            "image": "docker.xxx.io/user/user",
            "ports": [
              {
                "name": "http",
                "containerPort": 8080,
                "protocol": "TCP"
              }
            ],
            "resources": {
              "limits": {
                "cpu": "8"
              },
              "requests": {
                "cpu": "500m"
              }
            },
            "livenessProbe": {
              "httpGet": {
                "path": "/health",
                "port": 8080,
                "scheme": "HTTP"
              },
              "initialDelaySeconds": 600,
              "timeoutSeconds": 5,
              "periodSeconds": 5,
              "successThreshold": 1,
              "failureThreshold": 3
            },
            "readinessProbe": {
              "httpGet": {
                "path": "/health",
                "port": 8080,
                "scheme": "HTTP"
              },
              "initialDelaySeconds": 600,
              "timeoutSeconds": 5,
              "periodSeconds": 5,
              "successThreshold": 1,
              "failureThreshold": 3
            },
          }
        ]
      }
    }
  }
```
And can not find the service's route though "istioctl pc route"

**Expected behavior**
no warning and the pod is running well.

**Steps to reproduce the bug**

**Version (include the output of `istioctl version --remote` and `kubectl version`)**
istio-1.1.3
**How was Istio installed?**
helm templte and kubectl apply

**Environment where bug was observed (cloud vendor, OS, etc)**

**Affected product area (please put an X in all that apply)**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

Additionally, please consider attaching a [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) by attaching
the dump file to this issue.

**Comment by howardjohn** [comment]
Hey @wansuiye, we have made some recent changes for this type of issue. Can you try again on later versions (1.2 preferably, but the changes should be in 1.1.7+)?

If its still not working can you include the config that you have

**Comment by howardjohn** [comment]
Presumed fixed in 1.1.7+. Please update if this is not the case
